File: Daily Challenge/Java/Words Alignment - Top And Bottom 1.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine();
        String[] words = line.split(" ");
        int n = words.length;
        int maxLen = 0;

        for (String w : words)
            if (w.length() > maxLen) maxLen = w.length();

        String[] padded = new String[n];
        for (int i = 0; i < n; i++) {
            int pad = maxLen - words[i].length();
            if ((i + 1) % 2 == 1) {
                padded[i] = words[i] + "*".repeat(pad);
            } else {
                padded[i] = "*".repeat(pad) + words[i];
            }
        }

        for (int row = 0; row < maxLen; row++) {
            for (int col = 0; col < n; col++) {
                System.out.print(padded[col].charAt(row));
            }
            System.out.println();
        }
    }
}

File: Daily Challenge/C/Words Alignment - Top And Bottom 1.7.2025.c
#include <stdio.h>
#include <string.h>

int main() {
    char words[100][100];
    int n = 0, maxLen = 0;
    while (scanf("%s", words[n]) == 1) {
        int len = strlen(words[n]);
        if (len > maxLen) maxLen = len;
        n++;
        if (getchar() == '\n') break;
    }

    char padded[n][maxLen + 1];
    for (int i = 0; i < n; i++) {
        int len = strlen(words[i]);
        int pad = maxLen - len;
        if ((i + 1) % 2 == 1) {
            strcpy(padded[i], words[i]);
            for (int j = len; j < maxLen; j++) padded[i][j] = '*';
        } else {
            for (int j = 0; j < pad; j++) padded[i][j] = '*';
            strcpy(padded[i] + pad, words[i]);
        }
        padded[i][maxLen] = '\0';
    }

    for (int row = 0; row < maxLen; row++) {
        for (int col = 0; col < n; col++) {
            printf("%c", padded[col][row]);
        }
        printf("\n");
    }
    return 0;
}